mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -777,15 +778,34 @@ class ALSSuite
   }
 }
 
-class ALSCleanerSuite extends SparkFunSuite {
+class ALSCleanerSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private var localDir: File = _
+  private val conf = new SparkConf()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    localDir = Utils.createTempDir()
+    conf.set("spark.local.dir", localDir.getAbsolutePath)
+  }
+
+  override def afterEach(): Unit = {
+    try {
+      Utils.deleteRecursively(localDir)
+      // Once 'spark.local.dir' is set, it is cached. Unless this is manually cleared
+      // before/after a test, it could return the same directory even if this property
+      // is configured.
+      Utils.clearLocalRootDirs()
```

Comment:
I think I ask checking if it resolves the issue (not removed directories IIRC) to call `Utils.clearLocalRootDirs()` without all other changes after or before each test.